File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

	private static final int rate = 221;

	public double productionRatePerHour(int speed) {
		return speed * rate * successRate(speed);
	}

	public int workingItemsPerMinute(int speed) {
		return (int) (productionRatePerHour(speed) / 60);
	}

	private double successRate(int speed) {
		if (speed >= 1 && speed <= 4) {
			return 1;
		} else if (speed <= 8) {
			return 0.9;
		} else if (speed <= 9) {
			return 0.8;
		} else {
			return 0.77;
		}
	}
}

File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int) : float =
    match speed with
    | 0 -> 0
    | _ when speed < 5 -> 1
    | _ when speed < 9 -> 0.9
    | 9 -> 0.8
    | _ -> 0.77

let productionRatePerHour (speed: int) : float =
    (float speed) * (float 221) * (successRate speed)

let workingItemsPerMinute (speed: int) : int =
    int (productionRatePerHour speed / (float 60))
